src/main/java/me/juancarloscp52/entropy/mixin/BlockMixin.java

```diff
@@ -45,6 +49,25 @@
 @Mixin(Block.class)
 public class BlockMixin {
 
+    @Inject(method = "updateEntityMovementAfterFallOn", at = @At("HEAD"), cancellable = true)
+    private void bounce(BlockGetter level, Entity entity, CallbackInfo ci) {
+        if(Variables.bouncyBlocks && !entity.isSuppressingBounce()) {
+            ci.cancel();
+            Vec3 vec3 = entity.getDeltaMovement();
+            if (vec3.y < (double)0.0F) {
+                double d = entity instanceof LivingEntity ? (double) 1.0F : 0.8;
+                entity.setDeltaMovement(vec3.x, -vec3.y * d, vec3.z);
+            }
+        }
+    }
+
+    @Inject(method = "fallOn", at = @At("HEAD"), cancellable = true)
+    private void cancelFallDamage(Level level, BlockState state, BlockPos pos, Entity entity, double fallDistance, CallbackInfo ci) {
```

Comment:
Canceling fall damage in the block class like this will only affect blocks that do not override the `fallOn` method. Falling on pointed dripstone, hay blocks, and similar can still cause fall damage because of this. Perhaps injecting into the `Entity#causeFallDamage` method instead could be better.